pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -494,6 +504,52 @@ func (c *Cluster) saveMonConfig() error {
 	return nil
 }
 
+func (c *Cluster) saveCsiClusterConfig() error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      csiConfigName,
+			Namespace: c.Namespace,
+		},
+	}
+	k8sutil.SetOwnerRef(c.context.Clientset, c.Namespace, &configMap.ObjectMeta, &c.ownerRef)
+
+	type csiClusterConfig struct {
+		ClusterID string   `json:"clusterID"`
```

Comment:
Rook requires only a single ceph cluster per namespace, so the namespace where the configmap lives should be sufficient to match this ceph cluster and shouldn't need the clusterID.

Reply:
ceph csi requires a "clusterID" field for uniquely identifying the configuration to use (the subsection within the configmap). The way I understand it it must unique but does not literally have to be the cluster uuid, it just seems to be the most convenient and truly unique value to use. So we can't remove the key, but if you think you'd prefer a more human-freindly value such as the current namespace I had also considered that as value since the namespace has to be unique within the cluster.
The only other aspect worth considering is if the csi rook deploys would ever interact with other ceph clusters. In that case I might prefer to use the uuid as that really is going to be unique. But that might be too future looking.